### PR TITLE
ANS in Genesis

### PR DESCRIPTION
**Pull request by CapCap**
### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4805)

### CODE REVIEW

File: aptos-move/e2e-move-tests/src/tests/stake.rs [hunk 1]
```diff
@@ -81,33 +80,6 @@ fn test_staking_end_to_end() {
     assert_eq!(stake_pool.inactive, 0);
 }
 
-#[test]
-fn test_staking_mainnet() {
```
Comment (movekevin) [inline aptos-move/e2e-move-tests/src/tests/stake.rs:85]: This is an outdated test that doesn't use the right mainnet configuration. We have a new mainnet flow that's tested by a test in vm-genesis/src/lib.rs now

File: crates/aptos/src/genesis/tests.rs [hunk 2]
```diff
@@ -296,6 +304,10 @@ async fn create_layout_file(
     layout.chain_id = chain_id;
     layout.is_test = true;
     layout.total_supply = Some(INITIAL_BALANCE * 16);
+    if is_mainnet {
+        layout.ans_admin_address = Some(get_test_ans_funds_address());
```
Comment (CapCap) [inline crates/aptos/src/genesis/tests.rs:308]: `get_test_ans_admin_address`?

File: aptos-move/vm-genesis/src/lib.rs [hunk 3]
```diff
@@ -604,19 +641,15 @@ pub fn generate_genesis_change_set_for_testing(genesis_options: GenesisOptions)
         }
     };
 
-    generate_test_genesis(framework, Some(1)).0
-}
-
-/// Generate a genesis `ChangeSet` for mainnet
-pub fn generate_genesis_change_set_for_mainnet(genesis_options: GenesisOptions) -> ChangeSet {
```
Comment (movekevin) [inline aptos-move/vm-genesis/src/lib.rs:611]: Outdated mainnet genesis code that has been replaced with a different logic now.

File: aptos-move/e2e-tests/src/executor.rs [hunk 4]
```diff
@@ -122,11 +118,6 @@ impl FakeExecutor {
         Self::from_genesis(GENESIS_CHANGE_SET_TESTNET.clone().write_set())
     }
 
-    /// Creates an executor using the mainnet genesis.
-    pub fn from_mainnet_genesis() -> Self {
```
Comment (movekevin) [inline aptos-move/e2e-tests/src/executor.rs:126]: Outdated mainnet genesis code that has been replaced with a different logic now.

File: aptos-move/e2e-tests/src/executor.rs [hunk 5]
```diff
@@ -200,12 +191,6 @@ impl FakeExecutor {
         genesis
     }
 
-    /// Creates fresh genesis from the framework passed in.
-    pub fn custom_genesis(framework: &ReleaseBundle, validator_accounts: Option<usize>) -> Self {
```
Comment (movekevin) [inline aptos-move/e2e-tests/src/executor.rs:204]: Unused

File: aptos-move/vm-genesis/src/lib.rs [hunk 6]
```diff
@@ -701,9 +739,34 @@ impl TestValidator {
...
+    .unwrap()
+}
+
+pub fn get_test_ans_admin_address() -> AccountAddress {
+    AccountAddress::from_hex_literal(
+        "0x0ee16f0e4b47d5972f63a642385d52d301e53716b4e1fbd637b9a91a7f1979ba",
+    )
+    .unwrap()
+}
+
```
Comment (gregnazario) [inline aptos-move/vm-genesis/src/lib.rs:764]: Please remove

**Review by gregnazario — COMMENTED** [review]
This looks like it's doing a lot of weird hackery around the addresses and way more than I expected to be for just adding a couple addresses.

File: aptos-move/vm-genesis/src/lib.rs [hunk 7]
```diff
@@ -772,6 +819,19 @@ fn mainnet_genesis_config() -> GenesisConfiguration {
...
+                "0x3e89c7ef29468198fe58b3ced66d8c7dcb79b5f9fa27313464886334c35730e9",
+            )
+            .expect("Funds Address is valid"),
+        ),
+        ans_admin_address: Some(
+            AccountAddress::from_hex_literal(
+                "0x3e89c7ef29468198fe58b3ced66d8c7dcb79b5f9fa27313464886334c35730e9",
+            )
+            .expect("Admin Address is valid"),
+        ),
```
Comment (gregnazario) [inline aptos-move/vm-genesis/src/lib.rs:834]: Shouldn't these be the test values above?

File: crates/aptos/src/genesis/mod.rs [hunk 8]
```diff
@@ -225,7 +225,13 @@ pub fn fetch_mainnet_genesis_info(git_options: GitOptions) -> CliTypedResult<Mai
...
     )?;
 
-    let framework = client.get_framework()?;
+    let mut framework = client.get_framework()?;
+    framework.packages.iter_mut().for_each(|p| {
+        p.metadata
+            .modules
+            .iter_mut()
+            .for_each(|m| m.source = vec![])
+    });
```
Comment (gregnazario) [inline crates/aptos/src/genesis/mod.rs:234]: We should just build the framework bundle without sources rather than strip them here?

**Review by movekevin — APPROVED** [review]

**Pull request closed without merging**